public/components/datasources/components/new/configure_s3_datasource.tsx

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  EuiPanel,
+  EuiTitle,
+  EuiSpacer,
+  EuiText,
+  EuiLink,
+  EuiFormRow,
+  EuiFieldText,
+  EuiTextArea,
+  EuiButton,
+  EuiSelect,
+} from '@elastic/eui';
+import React, { useState } from 'react';
+import { OPENSEARCH_DOCUMENTATION_URL } from '../../../../../common/constants/data_connections';
+import { QueryPermissionsConfiguration } from '../manage/query_permissions';
+
+interface ConfigureS3DatasourceProps {
+  roles: Array<{ label: string }>;
+  selectedQueryPermissionRoles: Array<{ label: string }>;
+  setSelectedQueryPermissionRoles: React.Dispatch<React.SetStateAction<Array<{ label: string }>>>;
+  currentName: string;
+  currentDetails: string;
+  currentArn: string;
+  currentStore: string;
+  setStoreForRequest: React.Dispatch<React.SetStateAction<string>>;
+  setNameForRequest: React.Dispatch<React.SetStateAction<string>>;
+  setDetailsForRequest: React.Dispatch<React.SetStateAction<string>>;
+  setArnForRequest: React.Dispatch<React.SetStateAction<string>>;
+}
+
+export const ConfigureS3Datasource = (props: ConfigureS3DatasourceProps) => {
+  const {
+    setNameForRequest,
+    setDetailsForRequest,
+    setArnForRequest,
+    setStoreForRequest,
+    currentStore,
+    currentName,
+    currentDetails,
+    currentArn,
+    roles,
+    selectedQueryPermissionRoles,
+    setSelectedQueryPermissionRoles,
+  } = props;
+
+  const [name, setName] = useState(currentName);
+  const [details, setDetails] = useState(currentDetails);
+  const [arn, setArn] = useState(currentArn);
+  const [store, setStore] = useState(currentStore);
+  const authOptions = [
+    { value: 'option_one', text: 'No authentication' },
+    { value: 'option_two', text: 'SIGV4' },
+    { value: 'option_three', text: 'Basic Auth' },
+  ];
+  const [selectedAuthOption, setSelectedAuthOption] = useState(authOptions[0].value);
+
+  return (
+    <div>
+      <EuiPanel>
+        <EuiTitle>
+          <h1>{`Configure S3 Data Source`}</h1>
+        </EuiTitle>
+        <EuiSpacer size="s" />
+        <EuiText size="s" color="subdued">
+          {`Connect to S3with OpenSearch and OpenSearch Dashboards `}
+          <EuiLink external={true} href={OPENSEARCH_DOCUMENTATION_URL} target="blank">
+            Learn more
+          </EuiLink>
+        </EuiText>
+        <EuiSpacer />
+        <EuiText>
+          <h3>Data source details</h3>
+        </EuiText>
+        <EuiSpacer />
+        <EuiFormRow label="Data source name">
```

Comment:
Minor: do we want some type of EuiForm creator function here?

Reply:
If I am getting you correctly - I refactored it like so: [b7e2b9d](https://github.com/opensearch-project/dashboards-observability/pull/1049/commits/b7e2b9dff403457d4e93dfa2bedf72d32999b238). @mengweieric let me know if I was getting you.

Reply:
Sorry - figured out this requires a bigger refactor since its adding another level of nesting and state passing - will keep it like this for now